packages/vue-instantsearch/src/util/createInstantSearchComponent.js

```diff
@@ -23,10 +23,19 @@ export const createInstantSearchComponent = (component) =>
             'The `search-client` prop of `<ais-instant-search>` changed between renders, which may cause more search requests than necessary. If this is an unwanted behavior, please provide a stable reference: https://www.algolia.com/doc/api-reference/widgets/instantsearch/vue/#widget-param-search-client'
           );
 
-          this.instantSearchInstance.helper.setClient(searchClient).search();
+          this.instantSearchInstance.client = searchClient;
+          if (this.instantSearchInstance.mainHelper) {
+            this.instantSearchInstance.mainHelper
+              .setClient(searchClient)
+              .search();
+          }
         },
         indexName(indexName) {
-          this.instantSearchInstance.helper.setIndex(indexName || '').search();
+          if (this.instantSearchInstance.helper) {
+            this.instantSearchInstance.helper
+              .setIndex(indexName || '')
+              .search();
+          }
```

Comment:
Shouldn't this also manipulates `mainHelper` until we handle the case in instantsearch-core?